sycl/doc/extensions/GroupMask/GroupMask.asciidoc

```diff
@@ -178,64 +176,57 @@ work-item with the id `max_local_range()-1`.
 |`void flip(id<1> id)`
 |Toggle the value of the bit corresponding to the specified _id_.
 
-|`bool operator==(group_mask rhs) const`
+|`bool operator==(const group_mask &rhs) const`
 |Return true if each bit in this mask is equal to the corresponding bit in
  `rhs`.
 
-|`bool operator!=(group_mask rhs) const`
+|`bool operator!=(const group_mask &rhs) const`
 |Return true if any bit in this mask is not equal to the corresponding bit in
  `rhs`.
 
-|`group_mask operator &=(group_mask rhs)`
+|`group_mask &operator &=(const group_mask &rhs)`
 |Set the bits of this mask to the result of performing a bitwise AND with this
  mask and `rhs`.
 
-|`group_mask operator \|=(group_mask rhs)`
+|`group_mask &operator \|=(const group_mask &rhs)`
 |Set the bits of this mask to the result of performing a bitwise OR with this
  mask and `rhs`.
 
-|`group_mask operator ^=(group_mask rhs)`
+|`group_mask &operator ^=(const group_mask &rhs)`
 |Set the bits of this mask to the result of performing a bitwise XOR with this
  mask and `rhs`.
 
-|`group_mask operator pass:[<<=](size_t shift)`
+|`group_mask &operator pass:[<<=](size_t shift)`
 |Set the bits of this mask to the result of shifting its bits _shift_ positions
  to the left using a logical shift.  Bits that are shifted out to the left are
  discarded, and zeroes are shifted in from the right.
 
-|`group_mask operator >>=(size_t shift)`
+|`group_mask &operator >>=(size_t shift)`
 |Set the bits of this mask to the result of shifting its bits _shift_ positions
  to the right using a logical shift.  Bits that are shifted out to the right are
  discarded, and zeroes are shifted in from the left.
 
 |`group_mask operator ~() const`
 |Return a mask representing the result of flipping all the bits in this mask.
 
-|`group_mask operator <<(size_t shift)`
+|`group_mask operator <<(size_t shift) const`
 |Return a mask representing the result of shifting its bits _shift_ positions
  to the left using a logical shift.  Bits that are shifted out to the left are
  discarded, and zeroes are shifted in from the right.
 
-|`group_mask operator >>(size_t shift)`
+|`group_mask operator >>(size_t shift) const`
 |Return a mask representing the result of shifting its bits _shift_ positions
  to the right using a logical shift.  Bits that are shifted out to the right are
  discarded, and zeroes are shifted in from the left.
-|===
-
-|===
-|Function|Description
 
-|`group_mask operator &(const group_mask& lhs, const group_mask& rhs)`
-|Return a mask representing the result of performing a bitwise AND of `lhs` and
- `rhs`.
+|`group_mask operator &(const group_mask &rhs) const`
```

Comment:
I agree it was wrong. But I think this isn't the change we want. We should have all binary operators be friend members rather than members only taking rhs. This makes sure that the operator is correctly symmetric (e.g. for implicit conversion).

Reply:
Could you please provide an example of the use case for implicit conversion? We do not have any constructor for group_mask rather than copy constructor (we can create an object of that type using group_ballot() only).

Reply:
moved back to free friend functions